rastervision/backend/torch_utils/chip_classification/data.py

```diff
@@ -21,6 +21,19 @@ def build_databunch(data_dir, img_sz, batch_sz, class_names):
         train_dir, transform=aug_transform, classes=class_names)
     valid_ds = ImageFolder(valid_dir, transform=transform, classes=class_names)
 
+    if equalize:
+        train_ds.samples = equalize_classes_through_oversampling(
```

Comment:
You should be able to implement this by setting the sampler on the `Dataloader` -- that's what it's there for. Modifying the `Dataset` doesn't seem like the right approach. There is an example of how to do this here: https://github.com/azavea/raster-vision-fastai-plugin/blob/a0283b0f25ae805efaf2abdef2ef35ec390b0746/fastai_plugin/semantic_segmentation_backend.py#L110

Reply:
Yes, you are right, that is what we discussed in the issue. I'll see to it that the right changes are made.

Reply:
@lewfish A question regarding that. I first wanted to try a [WeightedRandomSampler](https://pytorch.org/docs/stable/data.html#torch.utils.data.Sampler), but this could lead to the situation where you cannot be sure that each image is each epoch for training. So say that we have the images of class A and B: [A1,A2,A3,B1], and weight vector [1,1,1,3], we want to train with [A1, A2, A3, B1, B1, B1], but because of the WeightedRandomSampler we could end up with [A1,A2,A1,B1,B1,B1], meaning we miss image A3. Or would you say that this effect in the long run would be minimal and can be disregarded? Or is there a way in which this effect can be reached with the samplers (I could not figure it out)?

Reply:
Each epoch it will pick a different random set of indices, so I doubt it matters. But there is also the `replacement` argument to https://pytorch.org/docs/stable/data.html#torch.utils.data.WeightedRandomSampler

Reply:
Yes, only the replacement (when set to false) will not allow you to draw more images than there are in the dataset, so this would allow you to undersample, not oversample. When set to True the above situation can occur. But Maybe you're right and its too much of an edge-case to really matter. Maybe it might only matter with few epochs and little data, which is what we're trying to avoid anyway. I'll get working on it with the WeightedRandomSampler and the weights as per your example.